### DISCUSSION TITLE
Terminal Failed to Open

### DISCUSSION THREAD

**Discussion by kulak** [cockpit-project/cockpit#19067] [Q&A]
I've got this error message and I am not sure where to start:

```
warning: No TTY for interactive shell (tcgetpgrp failed)
setpgid: Inappropriate ioctl for device
disconnected
```

It is Arch Linux installation from <https://archlinux.org/packages/extra/x86_64/cockpit/>

**Comment by jelly** [comment]
Do you get this in your browser console? What default shell do you use on your system?

**Reply by kulak** [reply]
The default shell of logged in user is `fish`.

**Reply by jelly** [reply]
Interesting, this is indeed a fish specific issue:

![image](https://github.com/cockpit-project/cockpit/assets/67428/b7a434ab-7de0-4599-a266-3a187bea204a)

Reproduced with:

```
[root@archlinux ~]# su admin
[admin@archlinux root]$ cd
[admin@archlinux ~]$ chsh
Changing shell for admin.
Password:
New shell [/bin/bash]: /bin/fish
Shell changed.
```

**Comment by allisonkarlitskaya** [comment]
Maybe we should be calling `login_tty()` from the preexec environment...

**Comment by allisonkarlitskaya** [comment]
Possible fix in https://github.com/cockpit-project/cockpit/pull/19076.

@kulak are you capable of testing this?

**Reply by kulak** [reply]
I might be able to.  Is there a quick reference on how it is done?

I have installed toolbox, but I am not sure what to do next.  I have not used toolbox.

I got to the point were I have installed locally `cockpit`, `toolbox`, cloned master and I was able to reproduce issue on my laptop. I would like to know, which commands to run to test changes.  The HACKING page is confusing and I don't know which part of it to apply.

**Reply by martinpitt** [reply]
The easiest way would be to [apply the patch](https://patch-diff.githubusercontent.com/raw/cockpit-project/cockpit/pull/19076.patch) directly to /usr/lib/python3.11/site-packages/hgext/convert/transport.py , either with `sudo patch` or with editing the file as root and adding the two new lines.
